### PR TITLE
Fix filter captions and showing filters with default value

### PR DESCRIPTION
**Pull request by siemiatj**
Properly selects and indicates active/not active filters, supports `defaultValue`, for not frequent filters allows showing captions for all selected filters instead of `Default` (which doesn't make sense). Plus a lot of cleanup and fixing data stored/sent to the server.
Related to #1944 & #1948

### CODE REVIEW

**Comment by siemiatj** [comment]
Can we turn off this codacy thing ? It's ultra retarted and doesn't help at all.

File: src/components/filters/FiltersItem.js [hunk 1]
```diff
@@ -20,8 +20,15 @@ class FiltersItem extends Component {
   constructor(props) {
     super(props);
 
+    const { active, data } = props;
```
[the remark this answers is not in the document]
Comment (siemiatj) [reply src/components/filters/FiltersItem.js]: wrong

File: src/components/filters/Filters.js [hunk 2]
```diff
@@ -2,32 +2,213 @@ import counterpart from 'counterpart';
...
-    filtersActive && this.init(filtersActive);
+    this.parseActiveFilters();
   }
 
-  init = filter => {
-    this.setState({
-      filter: filter,
+  // PARSING FILTERS ---------------------------------------------------------
+  parseActiveFilters = () => {
+    let { filtersActive, filterData } = this.props;
```
[the remark this answers is not in the document]
Comment (siemiatj) [reply src/components/filters/Filters.js]: go home codacy, you're drunk

**Comment by teosarca** [comment]
@siemiatj  
I think that `defaultValue` name is misleading. I think we shall read it as `inital value` and that's the value which shall be set to the field when user is opening the filter. And that's it.
It's just the initial value of that field. The filter shall not be send back (PATCH) to backend until user is not pressing the APPLY button.

More, sending `null` when user is clearing the field, for me it's correct.

**Comment by siemiatj** [comment]
@metas-mk Can you describe this case when filter is not applied ? Seems to be working fine : 
http://recordit.co/WbC5zPRqLX

**Comment by siemiatj** [comment]
Resetting to default should work now.

**Comment by siemiatj** [comment]
Ready to review.

**Pull request merged**